ProjectFiles/Assets/Scripts/Network/Client.cs

```diff
@@ -114,17 +96,18 @@ public void HandleNetworkEvents()
                     case NetworkEvent.Type.Connect:
                     {
                         Debug.Log($"Client: Successfully connected to {serverIP}:{serverPort}.");
-                        
-                        using (var writer = ClientNetworkEvent.ClientHandshake.GetWriter(0, Allocator.Temp))
+                        var handshake = new ClientHandshakeEvent();
+                        using (var writer = new DataStreamWriter(handshake.Length, Allocator.Temp))
                         {
+                            handshake.Serialise(writer);
```

Comment:
We do `event.serialise(writer)` a lot. And also create a writer for each case if the select.

Could we not just have a case which sets a variable called event 

And then after the case create a writer give it `event.Length`. Then call `event.Serialise`. 

Just removed the need to keep doing it in every case.

It was done like this for deserialising.

Reply:
I don't really understand this. What select?

Reply:
I think I see what you mean (I think), but when we're sending packets we're handling the concrete types anyway, since they're instantiated directly. I'm not sure what select you're referring to, though.

Reply:
Okay so this ones kinda my bad for trying to do it on a phone 🙈